wordpress-mobile/WordPress-Android
# BuildKite integration for Lint tasks

This PR adds BuildKite integration for Lint tasks. The changes are pretty straightforward and even though I did force-push fairly often while tinkering about, I still kept a lot of the commits, so the commit count is quite high. I wouldn't be opposed to squashing it all down to a single commit, but having some of the history here may prove helpful for me in the future.

[The docker image I opted to go with in this first iteration is fairly simple](https://github.com/wordpress-mobile/docker-android-build-image/blob/trunk/Dockerfile) and only contains android tooling and is about `614MB`. [I am experimenting with an image that contains a read-only dependency cache](https://github.com/wordpress-mobile/docker-android-build-image/blob/trunk/dependency-cache/Dockerfile), but I have not found that to make as big of a difference as I'd like for these tasks, so I am holding off on that for now.

The configuration phase for these builds take a lot of time which I'll address in a separate PR. We knew about this for some time, but since we were caching the configurations in CircleCI, we didn't need to address it yet, so it was in my todo list. Now that we use a fresh environment, it's not only necessary to do this change, it's also easier to explain the difference in a PR description with concrete numbers.

This also introduces a remote build cache which will only be usable in BuildKite builds because we are not allowing outside access to the build cache machine for now. (for security) In any other CI environment, it'll just report that it can't use the remote cache, but otherwise it'll have no impact.

Finally, I've updated the default jvm memory for Gradle to 6G to simplify the BuildKite integration. We already replace this setting in CircleCI builds and I'll propose to remove this from the example properties soon, so we can set it in an environment variable depending on the machine. Our current EC2 instances have 16G memory, but I think that'd be a bit too high to set to in the example properties.

Note that I'll remove CircleCI Lint task in a subsequent PR. I wanted to separate it for easier revert if it was necessary.

**To Test:**
* Check that the build passes in BuildKite - I have not enabled status reporting for the repository yet. I'll do it as soon as this PR is merged, so there aren't false positives in other's PRs.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

**PR submission checklist:**
